### PR TITLE
[SPARK-36270][BUILD][3.0] Change memory settings for enabling GA

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

Trying to adjust build memory settings and serial execution to re-enable GA.

### Why are the changes needed?

GA tests are failed recently due to return code 137. We need to adjust build settings to make GA work.

### Does this PR introduce _any_ user-facing change?

No, dev only.

### How was this patch tested?

GA

### CODE REVIEW

File: python/docs/conf.py [hunk 1]
```diff
@@ -335,7 +335,7 @@
 # If false, no index is generated.
 #epub_use_index = True
 def setup(app):
-    app.add_javascript('copybutton.js')
+    app.add_js_file('copybutton.js')
```
Comment (viirya) [inline python/docs/conf.py:338]: `add_javascript` was renamed to `add_js_file` in later sphinx version.

https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file

Let me know if you prefer to have separate PR for it.

**Comment by viirya** [comment]
retest this please

File: python/docs/conf.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/docs/conf.py:338]: I think that's fine

**Review by HyukjinKwon — APPROVED** [review]
Looks fine. We could skip/ignore some tests for now if there are few.

**Comment by viirya** [comment]
`org.apache.spark.sql.hive.thriftserver.SparkSQLEnvSuite` failed.

More seriously,  Jenkins has `java.lang.OutOfMemoryError: Metaspace`. I think we better to keep Jenkins work as before.

Let me try to adjust it a bit.

**Comment by viirya** [comment]
Jenkins test passed now. But `org.apache.spark.sql.hive.thriftserver.SparkSQLEnvSuite` continues to fail in GA:

```
[info] - SPARK-29604 external listeners should be initialized with Spark classloader *** FAILED *** (34 seconds, 465 milliseconds)
[info]   scala.Predef.refArrayOps[org.apache.spark.sql.util.QueryExecutionListener](session.get.listenerManager.listListeners()).exists(((x$1: org.apache.spark.sql.util.QueryExecutionListener) => x$1.isInstanceOf[test.custom.listener.DummyQueryExecutionListener])) was false (SparkSQLEnvSuite.scala:57)
```

It looks not a memory issue. I also ran it locally and it passed. Not sure why GA is special for the test.

**Comment by viirya** [comment]
I marked the test as ignored test for now. Need to investigate why it fails in GA.

**Comment by viirya** [comment]
I created #33511 to see if the test in current branch-3.0 can pass in GA or not.

**Comment by viirya** [comment]
Hm? Seems like GA works well in branch 3.0? If so, maybe we don't need backport this.

**Comment by viirya** [comment]
As GA can pass on branch 3.0. We don't need backport. Close this.

**Pull request closed without merging**